storage/innobase/include/dict0load.h

```diff
@@ -213,4 +213,6 @@ dict_process_sys_foreign_col_rec(
 					in referenced table */
 	ulint*		pos);		/*!< out: column position */
 
+dberr_t dict_get_sys_tablespace_indexes(
+  std::vector<uint32_t> &sys_root_pages);
```

Comment:
Can we define this function `static` in the compilation unit that needs it? Maybe there should be a separate compilation unit for system tablespace shrinking? All those functions could be declared `ATTRIBUTE_COLD`.